File: python/055. Jump Game.py
#coding=utf-8
__author__ = 'xuxuan'
class Solution(object):
    def canJump(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        nums[-1]=True
        index=len(nums)-2
        if index<0:
            return True
        while index>=0:
            length=nums[index]
            if index+length>=len(nums)-1:
                nums[index]=True
            else:
                flag=False
                for i in range(1,length+1):
                    if nums[index+i]:
                        flag=True
                        break
                nums[index]=flag
            index-=1
        return nums[0]

File: java/055. Jump Game.java
public class Solution {
    public boolean canJump(int[] nums) {
        int k=-1;
        int tar=-1;
        for (int i = nums.length-1; i>=0 ; i--) {
            if (k==-1&&nums[i]==0) {
                k=0;
                tar=i;
            }
            if (k==0) {
                if(tar==nums.length-1&&nums[i]>=tar-i){
                    k=-1;
                }
                if (nums[i]>tar-i) {
                    k=-1;
                }
            }
        }
        if (k==-1) {
            return true;
        }else {
            return false;
        }
    }
}